src/boosting/gbdt.cpp

```diff
@@ -265,6 +265,8 @@ void GBDT::Train(int snapshot_freq, const std::string& model_output_path) {
   Common::FunctionTimer fun_timer("GBDT::Train", global_timer);
   bool is_finished = false;
   auto start_time = std::chrono::steady_clock::now();
+  //pre-training metrics
+  OutputMetric(0);
```

Comment:
Just curious what OutputMetric(0) does? I know it's supposed to print the pre-training metrics. Guess more curious about how it works?